create-astronomical-data

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+from datetime import date
+from gzip import open
+from json import dumps
+
+from workalendar.astronomy import calculate_equinoxes, solar_term
+
+# In 2021, Skyfield cannot compute ephemeris after date 2053-10-08 23:58:51Z UT
+YEAR_INTERVAL = 30
+TIME_ZONES = (
+    'America/Santiago',
+    'Asia/Hong_Kong',
+    'Asia/Taipei',
+    'Asia/Tokyo',
+)
+
+
+if __name__ == '__main__':
+    current_year = date.today().year
+    first_year = current_year - YEAR_INTERVAL
+    last_year = current_year + YEAR_INTERVAL
+    years = range(first_year, last_year + 1)
+    equinoxes = dict()
+    solar_terms = dict()
+    n_years = len(years)
+    n_max = len(TIME_ZONES) * n_years
+    global_i = 0
+    for time_zone in TIME_ZONES:
+        print(f"{time_zone}")
+        equinoxes[time_zone] = dict()
+        equinoxes_tz = equinoxes[time_zone]
+        solar_terms[time_zone] = dict()
+        solar_terms_tz = solar_terms[time_zone]
+        for i, year in enumerate(years):
+            tz_pct = i * 100 // n_years
+            global_i += 1
+            global_pct = global_i * 100 // n_max
+            print(f"  {year}: {tz_pct}% ({global_pct}%)")
```

Comment:
This might be a opportunity to use https://github.com/tqdm/tqdm, a very nice progress bar lib :)

Reply:
I know the lib, but don’t see the point of adding one more dependency for something that will be updated once per release. I will do another commit so it will be an option on the table ;-)